bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/connection/PhilipsAirAPIConnection.java

```diff
@@ -0,0 +1,242 @@
+                case UNAUTHORIZED_401:
+                case NOT_FOUND_404:
+                    logger.debug("Philips Air Purifier device responded with status code {}", httpStatus);
+                    throw new PhilipsAirAPIException(String.format("Error with status %d", httpStatus));
+                case TOO_MANY_REQUESTS_429:
+                    cooldownTimer = System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(5);
+                default:
+                    logger.debug("Philips Air Purifier device responded with status code {}", httpStatus);
+                    throw new PhilipsAirAPIException(String.format("Error with status %d", httpStatus));
+            }
+        } catch (ExecutionException e) {
+            String errorMessage = e.getLocalizedMessage();
+            logger.trace("Exception occurred during execution: {}", errorMessage, e);
+            throw new PhilipsAirAPIException(errorMessage, e.getCause());
+        } catch (InterruptedException | TimeoutException e) {
+            logger.debug("Exception occurred during execution: {}", e.getLocalizedMessage(), e);
+            throw new PhilipsAirAPIException(e.getLocalizedMessage(), e.getCause());
+        } catch (Exception e) {
+            logger.error("Unexpected exception occurred during execution: {}", e.getLocalizedMessage(), e);
+            throw new PhilipsAirAPIException(e.getLocalizedMessage(), e.getCause());
+        }
+    }
+
+    public String exchangeKeys() throws GeneralSecurityException, InterruptedException, TimeoutException,
+            ExecutionException, InvalidAlgorithmParameterException {
+        if (this.cipher == null) {
+            return null;
+        }
+
+        String url = buildURL(KEY_URL, config.getHost());
+        String data = "{\"diffie\":\"" + this.cipher.getApow() + "\"}";
+
+        String encodedContent = getResponse(url, PUT, data, false);
+        JsonObject encodedJson = gson.fromJson(encodedContent, JsonObject.class);
+        String key = encodedJson.get("key").getAsString();
+        String hellman = encodedJson.get("hellman").getAsString();
+        String aesKey = this.cipher.calculateKey(hellman, key);
+        config.setKey(aesKey);
+        return aesKey;
+    }
+
+    public PhilipsAirPurifierData sendCommand(String parameter, Object value)
+            throws IllegalBlockSizeException, BadPaddingException, UnsupportedEncodingException, InvalidKeyException,
+            NoSuchAlgorithmException, NoSuchPaddingException, InvalidAlgorithmParameterException {
+        String commandValue = null;
+        if (parameter.equals("om")) {
+            commandValue = "{\"om\":\"" + value + "\",\"mode\":\"M\"}";
+        } else if ((value instanceof String)) {
+            commandValue = "{\"" + parameter + "\":\"" + value + "\"}";
+        } else {
+            commandValue = "{\"" + parameter + "\":" + value + "}";
+        }
+
+        logger.info("{}", commandValue.toString());

```

Comment:
The toString is called automatically.
```suggestion
        logger.info("{}", commandValue);

```